httomo/method_wrappers/dezinging.py

```diff
@@ -48,11 +48,14 @@ def execute(self, block: T) -> T:
         # check if data needs to be transfered host <-> device
         block = self._transfer_data(block)
 
-        args = self._build_kwargs(self._transform_params(self._config_params), block)
-        # plug in the correct value for axis instead of auto. In case if axis is absent we take
-        # it equal to 0. This is to avoid failure, but (missing) template parameters should be
-        # tested in the yaml checker potentially
-        self._config_params["axis"] = args.get("axis", 0)
+        if "axis" in self.parameters:
```

Comment:
for that one I was quite [explicit](https://github.com/DiamondLightSource/httomo/pull/425/files#diff-dfbcc5d5a18bd1d59d6fdbea5528bdd1f6fdafb6c33d728c026bebfaa0cc3a77R55) that we should be with TomoPy that requires is. But I guess your approach is more general just in case if `axis` is coming back in `httomolibgpu`?

Reply:
Yeah, I wasn't sure if `axis` could potentially come back in the methods in httomolibgpu, so I did it this way.

By the way, I didn't realise that this issue was being fixed among the other issues in #425, so feel free to close this PR if you'd prefer to fix it there :slightly_smiling_face:

Reply:
Yeah, sorry things started to misbehave in different places so I got a bit chaotic fixing them, that is why this quite generically named [PR](https://github.com/DiamondLightSource/httomo/pull/425) exists.